src/components/dropdown/daikin-dropdown.ts

```diff
@@ -0,0 +1,426 @@
+    this,
+    this._onClickOutside
+  );
+
+  private _updateFormValue() {
+    this._internals.setFormValue(this.value);
+  }
+
+  private _onClickOutside(): void {
+    this.open = false;
+  }
+
+  private _reflectItemsAndLabel(): void {
+    const items = this._items;
+
+    const selectedItem = items.find((item) => item.value === this.value);
+    for (const item of items) {
+      item.selected = item === selectedItem;
+    }
+
+    this._hasSelectedItem = !!selectedItem;
+    this._selectedItemLabel = selectedItem?.textContent ?? "";
+  }
+
+  private _handleClick(): void {
+    this.open = !this.open;
+  }
+
+  private _searchItem(key: string): void {
+    const items = this._items.filter(
+      (item) => item.textContent?.[0].toLowerCase() === key
+    );
+    const activeElement = document.activeElement;
+    const focusedItemIndex = activeElement
+      ? items.findIndex((item) => item.contains(activeElement))
+      : -1;
+
+    // Focus on the first dropdown that is enabled.
+    for (
+      let index = focusedItemIndex + 1, i = 0;
+      i < items.length;
+      index += 1, i++
+    ) {
+      index %= items.length;
+      const item = items[index];
+
+      if (item.disabled) {
+        continue;
+      }
+
+      item.focus();
+      break;
+    }
+  }
+
+  private _handleKeyDownEscape() {
+    if (this.open) {
+      // Close
+      this.open = false;
+    } else {
+      // Clear selection
+      this.value = null;
+    }
+  }
+
+  private _moveFocus(moveOffset: 1 | -1): void {
+    const items = this._items;
+
+    // Get focused item if any
+    const activeElement = document.activeElement;
+    const focusedItemIndex = activeElement
+      ? items.findIndex((item) => item.contains(activeElement))
+      : -1;
+
+    // If there is no dropdown focused, do nothing.
+    if (focusedItemIndex === -1 && moveOffset === -1) {
+      return;
+    }
+
+    // Focus on the first dropdown that is enabled.
+    for (
+      let index = focusedItemIndex + moveOffset, i = 0;
+      i < items.length;
+      index += moveOffset, i++
+    ) {
+      index = (index + items.length) % items.length;
+      const item = items[index];
+
+      if (!this.open) {
+        this.open = true;
+      }
+
+      if (item.disabled) {
+        continue;
+      }
+
+      this.updateComplete.then(
+        () => {
+          item.focus();
+        },
+        () => {
+          // do nothing
+        }
+      );
+
+      break;
+    }
+  }
+
+  private _handleKeyDown(event: KeyboardEvent): void {
+    const firstCharacterKey = event.key.length === 1 ? event.key : undefined;
+    const operation = (
+      {
+        ArrowDown: "down",
+        ArrowUp: "up",
+        Escape: "close",
+      } as const
+    )[event.key];
+
+    if (!operation && !firstCharacterKey) {
+      return;
+    }
+
+    if (this.open && firstCharacterKey) {
+      this._searchItem(firstCharacterKey);
+      return;
+    }
```

Comment:
If the user enter a printable key, the dropdown will open. Then, search.

```suggestion
    const printableCharacter = event.key.length === 1 ? event.key : null;
    if (printableCharacter) {
      this._searchItem(printableCharacter);
      return;
    }

    const operation = (
      {
        ArrowDown: "down",
        ArrowUp: "up",
        Escape: "close",
      } as const
    )[event.key];
    if (!operation) {
      return;
    }
```

Reply:
@yodas7 
Can I use the printable keys even if the dropdown menu is not open?
Could you tell me the reference documents you used?

Reply:
@poetrainy 
[****](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/)

Reply:
@yodas7 
Thank you for teaching me. I didn't know about this specification.